### ISSUE TITLE
You need to also check response.ok

### ISSUE THREAD

**Issue by Jean-Luc-Picard-2021** [kamahen/swipl-server-js-client#6]
This [code ](https://github.com/kamahen/swipl-server-js-client/blob/master/static/simple_client.js#L21 )might have a status 404 and no error:
```
For example, this code:

function fetchFromServer(request, callback) {
    // callback should take a single arg, the response from the server,
    fetch(...)
        .then(response => response.json())
        .then(callback)
        .catch(err => {
            alert('***fetch ' + JSON.stringify(request) + ': ' + err) });
}

became:

async function fetchFromServer(path, request, callback) {
    // callback should take a single arg, the response from the server.
    try {
        const response = await fetch(...);
        callback(await response.json());
    } catch(err) {
        alert('***fetch ' + JSON.stringify(request) + ': ' + err);
    }
}

```
The correct idiom would be, only reviewing the missing response.ok check:
```
For example, this code:

function fetchFromServer(request, callback) {
    // callback should take a single arg, the response from the server,
    fetch(...)
        .then(response => {if (response.status === 200) {
      return response.json();
    } else {
      throw new Error('Something went wrong on API server!');
    }})
        .then(callback)
        .catch(err => {
            alert('***fetch ' + JSON.stringify(request) + ': ' + err) });
}

became:

async function fetchFromServer(path, request, callback) {
    // callback should take a single arg, the response from the server.
    try {
        const response = await fetch(...);
        if (response.status === 200) {
             callback(await response.json());
        } else {
             throw new Error('Something went wrong on API server!');
        }
    } catch(err) {
        alert('***fetch ' + JSON.stringify(request) + ': ' + err);
    }
}
```

**Comment by Jean-Luc-Picard-2021** [comment]
See also here, there are also slightly different ways to code it:

> **Do you know why we check for response.ok while using fetch**
> https://dev.to/myogeshchavan97/do-you-know-why-we-check-for-response-ok-while-using-fetch-1mkd

**Comment by kamahen** [comment]
Fixed with commit 04afebb30686e7f44a388a80e64447f113843ad4
Thank-you for pointing this out - it is also documented here (using the older `Promise` interface): https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
